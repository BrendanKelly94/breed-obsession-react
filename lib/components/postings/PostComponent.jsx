import React from 'react';

const PostComponent = ({ post , selectPost }) => {

  const postObject = {
    name: post.name,
    status: post.status,
    age: post.age,
    sex: post.sex,
    breed: post.breed,
    city: post.city,
    State: post.state,
    email: post.email,
    phone: post.phone,
    description: post.description,
    pics: post.pics,
    street: post.street,
    zip: post.zip
  };

  return (
    <div className = "container row posts">
      <div className = "img-container col-lg-4 col-md-4 col-sm-4 col-xs-12">
        <span className = "helper"></span>
        <img className = "thumbnail" src={post.thumbnail} alt = "thumbnail"/>
      </div>

      <div className = "col-lg-8 col-md-8 col-sm-8 col-xs-12">
        <div className = "row">
          <p className = "name">{post.name}</p>
        </div>
        <div className = "row details">
          <p> <b>Status:</b> {post.status} </p>
          <p> <b>Age:</b> {post.age} </p>
          <p> <b>Sex:</b> {post.sex} </p>
          <p> <b>Breed:</b> {post.breed} </p>
          <p> <b>Location:</b> {post.city + ', ' + post.state} </p>
          <button className = "btn btn-lg modal-button" onClick = {() => selectPost(postObject)}><p className = "modal-button-text">View</p></button>
        </div>
      </div>
    </div>
  );

};

export default PostComponent;
