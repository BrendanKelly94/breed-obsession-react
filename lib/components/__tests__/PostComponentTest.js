import React from 'react';
import PostComponent from '../postings/PostComponent.jsx';

import renderer from 'react-test-renderer';

describe('PostComponent', () => {

  const testProps = {
    post: {},
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <PostComponent
        {...testProps}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();

  });
});
