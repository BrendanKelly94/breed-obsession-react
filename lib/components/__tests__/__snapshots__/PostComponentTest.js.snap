// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostComponent renders correctly 1`] = `
<div
  className="container row posts"
>
  <div
    className="col-lg-4 col-md-4 col-sm-4 col-xs-12"
  >
    <img
      alt="thumbnail"
      src={undefined}
    />
  </div>
  <div
    className="col-lg-8 col-md-8 col-sm-8 col-xs-12"
  >
    <div
      className="row"
    >
      <p
        className="name"
      />
    </div>
    <div
      className="row details"
    >
      <p>
         
        <b>
          Status:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Age:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Sex:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Breed:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Location:
        </b>
         
         , 
         
      </p>
      <button
        className="btn btn-lg btn-outline-dark modal-button"
        onClick={[Function]}
      >
        View
      </button>
    </div>
  </div>
</div>
`;
